docs/docs/35-references/10-promotion-steps.md

```diff
@@ -300,6 +300,53 @@ steps:
 # Render manifests to ./out, commit, push, etc...
 ```
 
+### `delete`
+
+`delete` deletes the files or the directory,
+
+#### `delete` Configuration
+
+| Name      | Type | Required | Description                              |
+|-----------|------|----------|------------------------------------------|
+| `path`    | `string` | Y | Path to the file or directory to delete. |
+
+#### `delete` Example
+
+The most common usage of this step is to clean up unwanted files
+or directories before proceeding to subsequent steps in the promotion
+process. For instance, you might use the delete step to remove
+intermediate configurations or artifacts generated in earlier steps.
+
+Consider a scenario where a Stage combines content from two working trees
+to render Stage-specific manifests. If temporary files are generated during
+this process, you can use the delete step to clean them up:
+
+```yaml
+vars:
+- name: gitRepo
+  value: https://github.com/example/repo.git
+steps:
+- uses: git-clone
+  config:
+    repoURL: ${{ vars.gitRepo }}
+    checkout:
+    - commit: ${{ commitFrom(vars.gitRepo, warehouse("base")).ID }}
+      path: ./src
+    - commit: ${{ commitFrom(vars.gitRepo, warehouse(ctx.stage + "-overlay")).ID }}
+      path: ./overlay
+    - branch: stage/${{ ctx.stage }}
+      create: true
+      path: ./out
+- uses: copy
+  config:
+    inPath: ./overlay/stages/${{ ctx.stage }}/kustomization.yaml
+    outPath: ./src/stages/${{ ctx.stage }}/kustomization.yaml
+- uses: delete
+  config:
+    path: ./overlay/stages/${{ ctx.stage }}/temp-file.yaml
+# Render manifests to ./out, commit, push, etc...
+```
+
 ### `kustomize-set-image`
```

Comment:
I'm trying to think of a more realistic example. I've noticed a tendency among users to put too much stock in examples, supposing them to be recommended patterns, which makes it important for the examples to be relatable and realistic...

At the same time, I don't think I've personally run into a need for this step. Looking back to #2860, the original issue was addressable using the existing git-clear step, so unfortunately, there is no inspiration to be found there.

I have absolutely no doubt that this step is generally useful and a completely necessary complement to the copy step... just need to find a realistic example for showcasing it.

I'll think on this a bit and if you don't mind, I'll update the example directly when I come up with something.

Ideas are welcome, of course.

Reply:
Other than this, docs look great. We're at like 99% on this PR.

Reply:
See https://github.com/akuity/kargo/pull/3086#discussion_r1893250643